docs/report_stage_iii.md

```diff
@@ -309,144 +334,257 @@ Calculating the averages, we get the following alternative gas cost schedule:
+PUSH19|3|6.00|99.84%|39.74%
+PUSH20|3|6.07|102.24%|41.48%
+PUSH21|3|6.17|105.55%|42.24%
+PUSH22|3|6.34|111.30%|42.33%
+PUSH23|3|6.52|117.21%|42.26%
+PUSH24|3|6.57|118.90%|43.29%
+PUSH25|3|6.66|122.16%|43.43%
+PUSH26|3|6.96|131.91%|43.82%
+PUSH27|3|7.01|133.50%|44.21%
+PUSH28|3|7.12|137.48%|44.33%
+PUSH29|3|7.24|141.33%|45.69%
+PUSH30|3|7.33|144.36%|46.09%
+PUSH31|3|7.59|153.01%|45.58%
+PUSH32|3|7.59|152.95%|47.23%
+DUP1|3|1.81|-39.71%|17.04%
+DUP2|3|1.78|-40.61%|19.33%
+DUP3|3|1.73|-42.25%|18.21%
+DUP4|3|1.71|-42.97%|18.88%
+DUP5|3|1.70|-43.25%|19.64%
+DUP6|3|1.78|-40.73%|19.07%
+DUP7|3|1.61|-46.28%|14.61%
+DUP8|3|1.59|-46.86%|14.48%
+DUP9|3|1.77|-41.09%|18.24%
+DUP10|3|1.69|-43.68%|17.61%
+DUP11|3|1.60|-46.80%|15.86%
+DUP12|3|1.78|-40.63%|16.66%
+DUP13|3|1.63|-45.65%|15.49%
+DUP14|3|1.63|-45.56%|14.09%
+DUP15|3|1.81|-39.59%|17.36%
+DUP16|3|1.62|-45.85%|14.24%
+SWAP1|3|2.03|-32.33%|15.97%
+SWAP2|3|2.06|-31.29%|14.43%
+SWAP3|3|2.06|-31.50%|14.16%
+SWAP4|3|2.04|-31.98%|14.78%
+SWAP5|3|2.01|-32.85%|16.43%
+SWAP6|3|2.11|-29.70%|16.35%
+SWAP7|3|2.25|-24.87%|17.06%
+SWAP8|3|2.03|-32.24%|14.60%
+SWAP9|3|2.01|-32.99%|16.21%
+SWAP10|3|2.37|-21.06%|19.15%
+SWAP11|3|1.99|-33.59%|15.01%
+SWAP12|3|2.12|-29.19%|15.98%
+SWAP13|3|2.13|-29.03%|16.78%
+SWAP14|3|1.97|-34.46%|15.40%
+SWAP15|3|2.38|-20.72%|21.62%
+SWAP16|3|2.29|-23.70%|19.30%
+
+
+> **Remarks**
 >
 > The data clearly shows that some changes can be made to the current gas proposal. Still, any such changes should be carefully considered, as there are other factors not taken into account in this research: ease of implementation, backward compatibility, hard fork requirement, existing tooling, hardcoded values in auxiliary software, etc.
 >
 >The alternative gas cost schedule is a proposal and should be treated as such. We believe that the methodology presented in this report is a good starting point for further research and discussion.
+
+
+### EXP
+For the detailed analysis of the `EXP` opcode, we have used the following data:
+- [EvmOne](./report_stage_iii_assets/evmone_measure_arguments_single_exp.html)
+- [Geth](./report_stage_iii_assets/geth_measure_arguments_single_exp.html)
+- [Netehrmind](./report_stage_iii_assets/nethermind_measure_arguments_single_exp.html)
+- [EtherumJS](./report_stage_iii_assets/ethereumjs_measure_arguments_single_exp.html)
+- [Erigon](./report_stage_iii_assets/erigon_measure_arguments_single_exp.html)
+- [Besu](./report_stage_iii_assets/besu_measure_arguments_single_exp.html)
+- [Rust EVM](./report_stage_iii_assets/revm_measure_arguments_single_exp.html)
+
+Please see 'EXP' section in the corresponding files.
+
+The `EXP` opcode proved quite difficult to estimate due to high client variability. Additionally, the measure marginal method was not able to sufficiently estimate the base cost of the opcode. The `EXP` measure marginal method used the argument size of 1 byte, as this provided the most consistent results. Still, for some clients, the measured cost was below 0.
```

Comment:
"Still, for some clients, the measured cost was below 0." This the outcome of the measure arguments method, not the measure marginal method as suggested.